pkg/kubelet/pluginmanager/pluginwatcher/README.md

```diff
@@ -22,6 +22,46 @@ should end with a DNS domain that is unique for the plugin. Each time a plugin
 starts, it has to delete old sockets if they exist and listen anew under the
 same filename.
 
+## Monitoring Plugin Connection
+
+The Kubelet monitors the gRPC connection to a plugin's **service socket** using a
+[gRPC stats handler](https://github.com/grpc/grpc-go/blob/master/examples/features/stats_monitoring/README.md).
+
+This enables the Kubelet to:
+- Detect when the plugin process has crashed, exited, or restarted
+- Deregister the plugin immediately when the connection is lost
+- Trigger cleanup of the plugin’s resources
+
+**registration socket** is monitored by the plugin manager. It registers the
+plugin when registration socket is created by the plugin and deregisters the
+plugin when socket is removed. After registration, the plugin begins serving
+gRPC requests over the **service socket**. Monitoring this service socket is
+therefore more accurate for detecting the real availability of the plugin.
```

Comment:
```suggestion
The **registration socket** is monitored by the plugin manager. It registers the
plugin when registration socket is created by the plugin and
the GetInfo gRPC call succeeds.
It  deregisters the plugin when the socket is removed.

The plugin should be ready to handle
gRPC requests over the **service socket** that
it returned in response to the GetInfo call because
the kubelet might try to use the service immediately.
Monitoring this service socket is
therefore more accurate for detecting the real availability of the plugin.
```